Mrc-6398 Manage access user help guide (#223)
mrc-ide/packit: app/public/docs/manageAccessHelpPage.md

> # Access Management Guide

Author: sadly not... but after we update project to vite we can use mdx which should allow this

## Overview

Packit uses a role-based access control (RBAC) system that provides flexible, fine-grained control over user permissions. The system supports both role-based permissions and direct user permissions, with the ability to apply permissions globally or scope them to specific resources.

> ## Core Concepts

Reviewer: Obviously doesn't need to be as wordy as IBM!

> ## Core Concepts

Author: have added note on permissions

### Users and Roles

- **Roles** contain permissions that are automatically inherited by all assigned users
- **Users** can have direct permissions in addition to those inherited from their roles

> - **Permission Inheritance** combines all role permissions with direct user permissions, with the most permissive scope taking precedence

Reviewer: or should this say 'permission resolution', the name of the next subheading?

These permissions can be applied globally or restricted to specific resources using scope qualifiers.

#### `packet.read`

- **Purpose**: Grants read access to packets with flexible scoping options.
- **Capabilities**:
  - View packet details
  - Access packet metadata
  - View packet groups and their contents

**Scope Examples:**

- `packet.read` - Global read access to all packets
- `packet.read:packetGroup:group1` - Read access to all packets in group1
- `packet.read:packet:group1:id` - Read access to specific packet with ID "id" in group1
- `packet.read:tag:tag2` _(coming soon)_ - Read access to packets tagged with "tag2"

#### `packet.manage`

- **Purpose**: Allows management of read access permissions for packets.

- **Capabilities:**

  - Grant read access to users or roles for packet groups or individual packets
  - Revoke read access from users or roles
  - Automatically includes `packet.read` for managed resources

**Scope Examples:**

- `packet.manage` - Can manage read access for any packet
- `packet.manage:packetGroup:group1` - Can manage read access for group1 packets only
- `packet.manage:packet:group1:id` - Can manage read access for one specific packet

> - `packet.manage:tag:tag2` _(coming soon)_ - Can manage read access for tagged packets

Reviewer: ```suggestion
- `packet.manage:tag:tag2` _(coming soon)_ - Can manage read access for packets with the tag 'tag2'
```

### Permission Dependencies

Some permissions automatically include others:

- `outpack.write` → includes `outpack.read`
- `user.manage` → includes global `packet.read`
- `packet.manage` → includes `packet.read` for the same scope

## Managing Access

### Creating Roles

1. Navigate to **Manage Roles**
2. Click **Add Role**
3. Enter role name and description

> 4. Select appropriate global permissions

Reviewer: Might be worth updating the form help text while we're at it - it currently lists the _global_ permissions and says 'Select the permissions that will be assigned to this role' which feels like it implies those are the only permissions you can choose from. I think it should say something like 'Select the global permissions to assign to this role, if any'

5. Click **Save**
6. Assign users and scoped permissions as needed

### Updating Role Assignments

**To modify users assigned to a role:**

1. In the Roles table, click the **⋯** (three dots) menu for the desired role
2. Select **Update Users**
3. Add or remove users as needed
4. Save changes

**To modify role permissions:**

1. In the Roles table, click the **⋯** menu for the desired role
2. Select **Update Permissions**
3. Add or remove permissions as needed
4. Save changes

### Managing User Access

**To assign roles to a user:**

1. In the Users table, click the **⋯** menu for the desired user
2. Select **Update Roles**
3. Add or remove role assignments
4. Save changes

**To assign direct permissions to a user:**

1. In the Users table, click the **⋯** menu for the desired user
2. Select **Update Permissions**
3. Add or remove specific permissions
4. Save changes

## Troubleshooting

### Permission Debugging

If a user cannot access expected resources:

1. **Check role permissions** - Verify all assigned roles have the necessary permissions

> 2. **Stale Permissions** - If a user recently had permissions changed, ensure they refresh their page to load the latest permissions

Reviewer: ```suggestion
2. **Stale permissions** - If a user recently had permissions changed, ensure they refresh their page to load the latest permissions
```

# Access Management Guide

## Overview

Packit uses a role-based access control (RBAC) system that provides flexible, fine-grained control over user permissions. The system supports both role-based permissions and direct user permissions, with the ability to apply permissions globally or scope them to specific resources.

## Core Concepts

### Roles, Permissions, and Users

- **Permissions** define what actions users can perform within the application.
- **Roles** contain permissions that are automatically inherited by all assigned users
- **Users** can have direct permissions in addition to those inherited from their roles

### Permission Resolution

When determining a user's effective permissions, the system evaluates:

1. **Role permissions** - All permissions from assigned roles
2. **Direct user permissions** - Specific permissions assigned directly to the user

> 3. **Scope precedence** - Global permissions override scoped permissions for the same action

Author: ive added back that statement

1. Select **Update users**
1. Add or remove users as needed
1. Save changes

**To modify role permissions:**

1. Navigate to **Manage Roles**
1. In the Roles table, click the **⋯** menu for the desired role
1. Select **Update permissions**
1. Add or remove permissions as needed
1. Save changes

### Managing User Access

**To modify a user's roles:**

1. Navigate to **Manage Users**
1. In the Users table, click the **⋯** menu for the desired user
1. Select **Update roles**
1. Add or remove role assignments
1. Save changes

**To modify a user's permissions:**

1. Navigate to **Manage Users**
1. In the Users table, click the **⋯** menu for the desired user
1. Select **Update permissions**
1. Add or remove specific permissions
1. Save changes

## Permission Types

### Global Permissions

These permissions apply across the entire application:

#### `packet.run`

- **Purpose**: Run packet groups to create new packets
- **Capabilities**:
  - Access the packet runner interface
  - Run any packet groups
  - View logs of packet runs

#### `outpack.read` and `outpack.write`

- **Purpose**: Direct packet metadata access
- **Capabilities**:

>   - `outpack.read`: Read all packet metadata

Reviewer: ```suggestion
  - `outpack.read`: Read all packet data and metadata
```
